manifests/init.pp

```diff
@@ -64,8 +67,9 @@
   validate_hash($settings)
   validate_hash($ppas)
 
-  if $proxy['host'] {
+  if $proxy['ensure'] or $proxy['host'] {
```

Comment:
I don't think this is going to behave the way you want it to. You'll end up with a weird proxy file if you have ensure != absent and the proxy host isn't specified.